sqlalchemy-stubs/engine/mock.pyi

```diff
@@ -1,22 +1,32 @@
 from typing import Any
+from typing import NoReturn
+from typing import TypeVar
 
-from . import base as base
-from .. import util as util
-from ..sql import ddl as ddl
+from . import base
+from . import interfaces
+from ..sql.compiler import Compiled
+
+_TMockConnection = TypeVar("_TMockConnection", bound=MockConnection)
 
 class MockConnection(base.Connectable):
-    def __init__(self, dialect: Any, execute: Any) -> None: ...
-    engine: Any = ...
-    dialect: Any = ...
-    name: Any = ...
-    def schema_for_object(self, obj: Any): ...
-    def connect(self, **kwargs: Any): ...
-    def execution_options(self, **kw: Any): ...
-    def compiler(self, statement: Any, parameters: Any, **kwargs: Any): ...
+    def __init__(self, dialect: interfaces.Dialect, execute: Any) -> None: ...
```

Comment:
execute seems a callable like this https://github.com/sqlalchemy/sqlalchemy/blob/f8b3a6a6e8a51bcb697ffd9aa79ad6ee706e1d24/lib/sqlalchemy/engine/mock.py#L80 also below in the create mock engine.

but I think the usage of this code is very low, so we can merge as is and maybe update later

Reply:
Yeah, I didn't get too worried about this file since it doesn't seem to be used too heavily